core/src/main/java/nu/yona/server/crypto/CryptoSession.java

```diff
@@ -30,21 +28,16 @@
 
 public class CryptoSession implements AutoCloseable
 {
+	static final byte CURRENT_CRYPTO_VARIANT_NUMBER = 1;
 	private static final String CIPHER_TYPE = "AES/CBC/PKCS5Padding";
-
-	public interface Executable<T>
-	{
-		T execute();
-	}
-
-	public interface VoidPredicate
-	{
-		boolean test();
-	}
+	private static final String SECRET_KEY_ALGORITHM = "PBKDF2WithHmacSHA256";
+	private static final int SECRET_KEY_LENGTH_BITS = 128;
+	/**
+	 * As we are using secure random passwords, a fixed salt suffices.
```

Comment:
You could as a hacker try a lot of random passwords with this fixed salt appended. The chance that you hit a match is low. But it is higher than trying to find a combination of user ID and password. So I think it is acceptible but decreases security a little.